radio/src/main.cpp

```diff
@@ -500,6 +500,10 @@ void perMain()
 
   event_t evt = getEvent(false);
 
+#if defined(HARDWARE_TOUCH)
```

Comment:
Maybe accademical, but I guess it should be #if defined(HARDWARE_TOUCH) && defined(LUA) ? A couple of other places too

Reply:
interesting point. So far HARDWARE_TOUCH was independent on LUA. I would suspect that this would be so also on the long run. Also, the changes embraced in these defines basically "just" took out the touch init & read from the radio_diaganas and moved it to more generic places, and did allow some touch testing. Again totally independent on the lua additions. My feeling is that not linking this to lua is probably a good thing.

Reply:
Well in a 2.3 codeline, we are not using touch, unless for this specific LUA scenario, so why read touch if you don't have LUA ? The touch was previously read only in diaganas for the same reason

Reply:
I think the point which wasn't clear to me was what you were explaining in the other comment, that it would be included by default and would not be an option

yet, I'm not fully understanding: touch in diaganas should be now also available only if also lua is available?

IMHO it would not make much sense to keep the old code in diaganas in case one would not use lua, I rather would consider it more clean to do it in diaganas as it is right now for both with and without lua - in which case we would be back to the defines as they are now...

Reply:
In diaganas, the touch panel is activated on entry in the screen, and desactiavted on exit. The only purpose of this existing was to allow buyers of the tx16s to reassure themselved confirm they actually have touch hardware, and it is working. We do not intend to have touch hardwrae running and polled if not usable, which is why TP is deinit on screen exit. In case of this PR, if LUA isn't specified, then touch becomes back not usable, and therefore should not be active, imho

Reply:
ok
so, if TOUCH and not LUA, then one should get the old behavior for diaganas (i.e. diaganas ints and reads touch)? Or not, becasue these users instead just could use the compile with LUA?
Just want to get it right :)

Reply:
I'm personnaly ok if after this PR the diaganas part is disabled if LUA is disabled, I think both ways are ok. LUA option is not an end user option in Companion (unlike LUAC), but there are professional users that have LUA disabled in their builds, it is mmore is this sceneario that I believe we should not be pooling the TP

Reply:
ok, even I understand it now :)
I'll do this evening

Reply:
I did it now not by adding && defined(LUA) to all these instances, but rathe by not setting HARDWARE_TOUCH in the CMakeFile if LUA is defined. This achieves exactly the same, but is more flexible and IMHO just nicer. Hope that's acceptable. For the moment, I consider this resolved.